lib/k8s-e2e-tests/tests/vector-agent.rs

```diff
@@ -189,7 +205,18 @@ async fn simple_raw_config() -> Result<(), Box<dyn std::error::Error>> {
         )
         .await?;
 
-    let mut log_reader = framework.logs(&namespace, &format!("daemonset/{}", override_name))?;
+    // Make sure we read the correct nodes logs.
+    let node = framework
+        .get_node_for_pod(&pod_namespace, "test-pod")
+        .await
+        .expect("need the node name");
+
+    let vector_pod = framework
+        .get_pod_on_node(&namespace, &node, &override_name)
+        .await
+        .expect("cant get the vector pod running on the test node");
+
+    let mut log_reader = framework.logs(&namespace, &format!("pod/{}", vector_pod))?;
```

Comment:
nit: it looks like this snippet is copy/pasted quite a number of times. Might make sense to move this into a helper.